types/src/transaction/encrypted_payload.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::transaction::{TransactionExecutable, TransactionExecutableRef, TransactionExtraConfig};
+use anyhow::{bail, Result};
+use aptos_crypto::HashValue;
+use serde::{Deserialize, Serialize};
+use std::ops::Deref;
+
+pub type CipherText = Vec<u8>;
+pub type EvalProof = Vec<u8>;
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub enum EncryptedPayload {
+    V1(EncryptedPayloadV1),
+}
+
+impl Deref for EncryptedPayload {
+    type Target = EncryptedPayloadV1;
+
+    fn deref(&self) -> &Self::Target {
+        let Self::V1(payload) = self;
+        payload
+    }
+}
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub enum EncryptedPayloadV1 {
```

Comment:
why it's nested enum?

Reply:
In case we wanted a V2 of the EncryptedPayload in the future. We can also make EncryptedPayloadV2 at the Payload level. is this too much nesting?

Reply:
I think its too much nesting, we can use the Payload enum to extend later if necessary. I removed the versioning and just used EncryptedPayload.